src/test/kotlin/org/opensearch/commons/replication/ReplicationPluginInterfaceTests.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.commons.replication
+
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.Answers
+import org.mockito.ArgumentMatchers
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.jupiter.MockitoExtension
+import org.opensearch.action.ActionType
+import org.opensearch.action.support.master.AcknowledgedResponse
+import org.opensearch.client.node.NodeClient
+import org.opensearch.commons.replication.action.StopIndexReplicationRequest
+import org.opensearch.core.action.ActionListener
+
+@Suppress("UNCHECKED_CAST")
+@ExtendWith(MockitoExtension::class)
+internal class ReplicationPluginInterfaceTests {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private lateinit var client: NodeClient
+
+    @Test
+    fun stopReplication() {
+        val request = Mockito.mock(StopIndexReplicationRequest::class.java)
+        val response = AcknowledgedResponse(true)
+        val listener: ActionListener<AcknowledgedResponse> =
+            Mockito.mock(ActionListener::class.java) as ActionListener<AcknowledgedResponse>
+
+        Mockito.doAnswer {
+            (it.getArgument(2) as ActionListener<AcknowledgedResponse>)
+                .onResponse(response)
+        }.whenever(client).execute(Mockito.any(ActionType::class.java), Mockito.any(), Mockito.any())
+
+        ReplicationPluginInterface.stopReplication(client, request, listener)
+        Mockito.verify(listener, Mockito.times(1)).onResponse(ArgumentMatchers.eq(response))

```

Comment:
not sure about what this test is checking, would you explain a bit

Reply:
Sure. I've written this test on similar lines as done in [NotificationsPluginInterfaceTests.kt]( https://github.com/opensearch-project/common-utils/blob/main/src/test/kotlin/org/opensearch/commons/notifications/NotificationsPluginInterfaceTests.kt#L63) and [AlertingPluginInterfaceTests.kt](https://github.com/opensearch-project/common-utils/blob/20f9ef6a315c0c51279cdcbe02e27597e3399afa/src/test/kotlin/org/opensearch/commons/alerting/AlertingPluginInterfaceTests.kt#L58)

- An AcknowledgedResponse response object is created with a value of true (expected response). 
- Have mocked the client object such that, when stopReplication() invokes client.execute(), it would trigger the onResponse method of the ActionListener with the successful **response** object created earlier.
- The test runs stopReplication() and verifies that the onResponse() method of the listener was called exactly once with the provided response object as an argument.
- This is to simulate a successful execution and verifies that the stopReplication method correctly handles the response.

Reply:
Thanks for the explain

What I feel weird is we mock the client.execute to do listener.onResponse
Then we call stopReplication which just does client.execute, and verify listener.onResponse called once

In the end, it seems just testing stopReplication does client.execute exactly once 🤔